### PR TITLE
delete target and fix reset-pass errors

### PR DESCRIPTION
**Pull request by brunomichetti**
I've had a problem with a branch in git so I had to fix tests for reset-password.
Besides, it has the feature: delete targets

### CODE REVIEW

File: targets/views.py [hunk 1]
```diff
@@ -1,50 +1,118 @@
...
+    permission_classes = (IsAuthenticated,)
+
+    def delete(self, request, target_id, format=None):
+        current_user = get_user_request(request)
+        if (current_user.is_authenticated and
+                Target.objects.filter(
+                    Q(user_id=current_user.id) & Q(id=target_id)
+                    )):
+            Target.objects.filter(id=target_id).delete()
+            return Response(status=status.HTTP_200_OK)
```
Comment (kaozdl) [inline targets/views.py]: The status code for deletions in Rest framework is HTTP_204_NO_CONTENT

File: targets/urls.py [hunk 2]
```diff
@@ -1,7 +1,9 @@
 from django.urls import path
 
-from targets.views import TargetView
+from targets.views import TargetView, TargetMatchView, TargetDeleteView
 
-urlpatterns = [    
+urlpatterns = [  
     path('', TargetView.as_view()),
+    path('matches/', TargetMatchView.as_view()),
+    path('<int:target_id>/', TargetDeleteView.as_view()),
```
Comment (kaozdl) [inline targets/urls.py]: You should use the same view for edit, delete and retrieve.

File: target/settings.py [hunk 3]
```diff
@@ -87,18 +88,20 @@
 # Password validation
 # https://docs.djangoproject.com/en/1.11/ref/settings/#auth-password-validators
 
+COMMON = 'django.contrib.auth.password_validation'
```
Comment (adebard) [inline target/settings.py]: What is the advantage of making this change? I never seen this before. Only for respect the character limit?

File: target/settings.py [hunk 4]
```diff
@@ -179,8 +176,7 @@
     from target.circleci_settings import *
 else:
     from target.settings import *
```
Comment (adebard) [inline target/settings.py]: Why are you importing target.settings, you are importing the same current settings. Here you only need to import the local settings.

File: target/settings.py (hunk 3 above)
Comment (kaozdl) [reply target/settings.py]: yes, it's only to respect the character limit.
What would be your advice in this kind of situations?

File: target/urls.py [hunk 5]
```diff
@@ -23,13 +23,18 @@
...
     path('admin/', admin.site.urls),
-    path('', include('django.contrib.auth.urls')),    
-    path('rest-auth/registration/account-confirm-email/<str:key>/', confirm_email),
-    path('rest-auth/registration/', include('rest_auth.registration.urls')),    
-    path('rest-auth/account-confirm-email/', VerifyEmailView.as_view(), name='account_email_verification_sent'),
-    path('rest-auth/password/reset/', PasswordResetView.as_view(), name='password_reset'),
+    path('', include('django.contrib.auth.urls')),
+    path(
+        'rest-auth/registration/account-confirm-email/<str:key>/',
+        confirm_email,
```
Comment (adebard) [inline target/urls.py]: Here there is an extra ","  please remove it, and the parenthesis should be at this line. Also would be better if you put the module 'rest-auth/registration/account-confirm-email/<str:key>/' at the same level of path, like path('rest-auth/registration/....

File: target/settings.py (hunk 4 above)
Comment (brunomichetti) [reply target/settings.py]: You're right!
I didn't note!
Thanks

File: target/urls.py (hunk 5 above)
Comment (brunomichetti) [reply target/urls.py]: If I do that, then vscode shows me a PEP8 warning: continuation line under-indented for visual indent
I'm going to look in PEP8 documentation to solve this

Comment (adebard) [reply target/urls.py]: Oh I see why you did that, ok in that case it's ok, just remove the extra ",". But anyway, some companies has his own rules, some companies decide to use 120 characters by line in python. Here we don't have any policy yet, but sometimes (I said sometimes, not always, just use common sense) you can be go more over than 80 characters in one line using common sense. As you can see, the code generated by Django sometimes goes over that limit. Following pep8 is just about maintaining consistency in your code. PEP8 is still a guideline, not a strict rule of law :)

File: target/settings.py (hunk 3 above)
Comment (adebard) [reply target/settings.py]: I have a similar discussion of character limit in other line.
https://github.com/brunomichetti/target-project/pull/12#discussion_r261716903
As I mention there, for that reason the code generated by Django sometimes goes over that limit.

File: target/urls.py (hunk 5 above)
Comment (brunomichetti) [reply target/urls.py]: I've fixed it!
I think I've found a good way to follow PEP8 and follow your recommendation

Comment (adebard) [reply target/urls.py]: Cool! 👍

**Pull request merged**